pippy/PipelineSchedule.py

```diff
@@ -236,6 +252,10 @@ def step_microbatches(
         # Delay send waits
         bwd_sends_to_wait: List[dist.Work] = []
         for i in range(self._n_microbatches):
+
+            # set library-specific data-parallel config flags to ensure gradient accumulation across microbatches
+            self._configure_data_parallel_mode(i == self._n_microbatches - 1)
+
```

Comment:
Following the above comment -- if we move `_configure_data_parallel_mode` to `PipelineStage` level, then `PipelineSchedule` can remain agnostic to submod type.